ci-operator/step-registry/ipi/install/ipi-install-chain.yaml

```diff
@@ -6,5 +6,6 @@ chain:
   - ref: ipi-install-hosted-loki
   - ref: ipi-install-install
   - ref: ipi-install-times-collection
+  - ref: multiarch-nodes-readiness
```

Comment:
`chain: ipi-install` is widely used by various platforms,  I'm thinking if adding a restriction to the platform is required.

Reply:
The same feeling, this probably bring some risk to break the existing huge number of ci jobs, so maybe we can  insert the step to the specific provision chain to leverage the risk.

Reply:
Hi, I see some value in this step to be run beyond the multiarch-specific verification we're addressing here.

The installer concludes the installation successfully when two worker nodes are ready, and the COs reconcile successfully. 

In some cases, especially with heterogeneous clusters, we've already faced escapes because there were enough ready nodes for the cluster to be considered healthy, but others weren't ready/available due to bugs.

Therefore, having a check before the `test` chain starts that verifies all the nodes are ready could benefit everyone. (We may have some other similar check later, e.g., in the cucushift steps, but not all the workflows use it)

Finally, I wouldn't specify this step in more specific chains, as that can be heavier to control.

I understand this change feels risky, though, as already anticipated in Slack. One option is to roll out this change only for the platforms that have introduced the separated `CONTROL_ARCH` and `COMPUTE_ARCH` Env variables (GCP and AWS - but only AWS will run it for all the jobs as they default to amd64).  

What do you think?

Reply:
> What do you think? See https://github.com/openshift/release/pull/54703/files#diff-b410185d62e12efa994a43dedaef9dae2ea830bbbcd813eb3b1701f2a980a981R7-R10

(I can not see this part of code.)

> but only AWS will run it for all the jobs

Sounds good to me,  but there is a simpler var that could be used: `CLUSTER_TYPE` https://github.com/openshift/release/blob/cb6980bf38e7fc9771a4f57db6e704d1c8dafe53/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L573 ,  WDYT?

Reply:
I didn't want to use CLUSTER_TYPE because it's actually a variable concerning the cluster profile being used rather than the actual 'cloud provider'.

Also, from an engineering perspective, it looks like creating a coupling between the logic of the proposed steps and variables (the cloud provider/cluster profile) that are not really related to what we are validating. 

As stated in https://github.com/openshift/release/pull/54703#discussion_r1698870796, it would be nice for `nodes-readiness` to run for all the installations consumed by the ipi-install chain, and that `multiarch-validate-nodes` is rolled out while we roll-out the support for mixarch at day0 ANDthe CI pivot to CONTROL_ARCH + COMPUTE_ARCH, deprecating OCP_ARCH.

Reply:
Resolving this comment as this is addressed by checking whether CONTROL_ARCH OR COMPUTE_ARCH are empty. I the future, we may change to AND (e.g., if we really deprecate OCP_ARCH).